Сетка страниц и крупных блоков на флексах

AlexPolyakovWeb opened the pull request:
Подготовительный этап перед началом работы с флексами

---
:mortar_board: [Сетка страниц и крупных блоков на флексах](https://up.htmlacademy.ru/htmlcss/33/user/1967959/tasks/7)

:boom: https://htmlacademy-htmlcss.github.io/1967959-sedona-33/5/

AlexPolyakovWeb committed:
Исправил line-height/подготовил  флекс контейнер main/добавил патерн стики-футера

AlexPolyakovWeb committed:
Добавил обёртки в хедере на главной

AlexPolyakovWeb committed:
Сделал навигацию в хедере

предлагаю двигаться по блокам ))) чтоб я не свернулся )))) от правок

AlexPolyakovWeb committed:
Спасай пожалуйста(

в вк оставил крик души

AlexPolyakovWeb committed:
хедер+первый экран

AlexPolyakovWeb committed:
Проект Sedona уперся в разметку блока преимуществ

4err commented on index.html:
```diff
@@ -6,199 +6,185 @@
+      </header>
+      <!--Первый экран-->
+      <main class="main-index">
+        <div class="main-container">
```
Эта обертка избыточна. У тебя уже есть общий центровщик, поэтому имеет смысл отталкиваться от него.

4err commented on index.html:
```diff
@@ -6,199 +6,185 @@
+            <h2 class="visual-hidden">Преимущества</h2>
+            <!--Нумерованный список преимуществ с вложенным списком-->
+            <ol class="edge-list">
+              <li class="edge-item edge-item-long-1">
```
Старайся не использовать нумерацию в классах. Через неделю ты не вспомнишь, чем отличаются разные номера.

4err commented on index.html:
```diff
@@ -6,199 +6,185 @@
+              <li class="edge-item edge-item-long-1">
+                <h3 class="edge-title long">Настоящий городок</h3>
+                <p class="edge-number long">-№1-</p>
+                <p class="edge-description-text long">Седона — не аттракцион для туристов,
```
Доп класс `long` тут избыточен. Ты можешь стилизовать эти элементы отталкиваясь от наличия класса `edge-item-long` в верхнем уровне.

4err commented on index.html:
```diff
@@ -6,199 +6,185 @@
+            <p class="section-description">Укажите предполагаемые даты поездки,
+              и мы покажем вам лучшие предложения гостиниц в Седоне
+            </p>
+            <div class="search-button-hotell">
```
Лишняя `l` в `hotel`

4err commented on styles/styles.css:
```diff
@@ -117,99 +149,109 @@ body {
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-end;
+  background-image: url("../images/mainPage/mainScrin/indexBackground.jpg");
```
В названиях путей и файлов не следует использовать заглавные буквы, пробелы и кириллицу.

4err reviewed (changes requested):
Не забывай сбрасывать отступы у именных блоков. Почти у каждого именного блока есть свои отступы, поэтому прежде чем что-то с ними делать нужно обнулить их. В частности сейчас у тебя у боди и списков они явно выражены. 

Касательно причин. Для того, чтобы все работало так как в макете, у тебя нехватает оберток. Обрати внимание на квадратные блоки - причины, еда, жилье. Они все одинаковые, кроме фона, соответственно у них должен быть одинаковый класс и обертка. Изображение же уже идет дополнительно следующим блоком, который находится рядом. Таким образом это позволит нам сформировать нужную сетку. 

Также рекомендую вот этот конструктор, чтобы проектировать и тренироваться, как работает то или иное свойство флексов:
https://the-echoplex.net/flexyboxes/

В нем можно спланировать нужную структуру и понять, чего не хватает. 

Скрытые заголовки можно уже скрыть.

AlexPolyakovWeb committed:
Убрал лишнюю обёртку/исправил наименования файлов/убрал лишний класс

AlexPolyakovWeb committed:
сверстал блок преимуществ

4err reviewed (changes requested):
Не, список разбивать нельзя, если делаешь их списком.  Это один большой список, у которого в некоторых элементах доп обертка. 

item3,4,5 нужно переименовать и привести в более осмысленный вид. Лучше ориентироваться по цвету фона. Зачем нам столько уникальных классов, если всего 3 разных цвета фона?

AlexPolyakovWeb committed:
сделал всё одним списком/убрал цифры в классах

AlexPolyakovWeb committed:
Update index.html

4err commented:
Хорошо. Продолжай

AlexPolyakovWeb committed:
сделал главную

не понял почему не загружается фоновое изображение в блоке подписки на главной

AlexPolyakovWeb committed:
сверстал каталог большие блоки

AlexPolyakovWeb committed:
немного подпилил

AlexPolyakovWeb committed:
подправил форму на главной

4err commented:
Сетку гостиниц тоже нужно сделать

AlexPolyakovWeb committed:
Сделал разметку карточек в каталоге

Не разобрался как можно добавить  свойство outline элементам для подсветки их размеров
